packages/cdktf/lib/tokens/private/resolve.ts

```diff
@@ -92,9 +118,10 @@ export function resolve(obj: any, options: IResolveOptions): any {
 
   // protect against cyclic references by limiting depth.
   if (prefix.length > 200) {
-    throw new Error(
-      "Unable to resolve object tree with circular reference. Path: " + pathName
-    );
+    throw new Error(`Unable to resolve object tree with circular reference at '${pathName}'.
+This error is thrown if the depth of the object tree is greater than 200 to protect against cyclic references.
+To resolve this inspect the construct creating the cyclic reference (most likely in '${pathName}') and make sure
+it does not create an infinite construct nesting.`);
```

Comment:
```suggestion
it does not create an infinite nesting of constructs.`);
```
nit, feel free to disregard